src/payment/strategies/googlepay/googlepay-stripe-upe-initializer.ts

```diff
@@ -0,0 +1,107 @@
+        hasShippingAddress: boolean
+    ): Promise<GooglePayPaymentDataRequestV2> {
+        return Promise.resolve(this._getGooglePayPaymentDataRequest(
+            checkout,
+            paymentMethod,
+            hasShippingAddress
+        ));
+    }
+
+    teardown(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    parseResponse(paymentData: GooglePaymentData): Promise<TokenizePayload> {
+        try {
+            const payload = JSON.parse(paymentData.paymentMethodData.tokenizationData.token);
+
+            return Promise.resolve({
+                nonce: payload.id,
+                type: payload.type,
+                details: {
+                    cardType: payload.card.brand,
+                    lastFour: payload.card.last4,
+                },
+            });
+        } catch (err) {
+            throw new InvalidArgumentError('Unable to parse response from Google Pay.');
+        }
+    }
+
+    private _getGooglePayPaymentDataRequest(
+        checkout: Checkout,
+        paymentMethod: PaymentMethod,
+        hasShippingAddress: boolean
+    ): GooglePayPaymentDataRequestV2 {
+        const {
+            outstandingBalance,
+            cart: {
+                currency: { code: currencyCode },
+            },
+        } = checkout;
+
+        const {
+            initializationData: {
+                googleMerchantName: merchantName,
+                googleMerchantId: merchantId,
+                platformToken: authJwt,
+                stripeVersion,
+                stripePublishableKey,
+                stripeConnectedAccount,
+            },
+            supportedCards,
+        } = paymentMethod;
+
+        return {
+            apiVersion: 2,
+            apiVersionMinor: 0,
+            merchantInfo: {
+                authJwt,
+                merchantId,
+                merchantName,
+            },
+            allowedPaymentMethods: [{
+                type: 'CARD',
+                parameters: {
+                    allowedAuthMethods: ['PAN_ONLY', 'CRYPTOGRAM_3DS'],
+                    allowedCardNetworks: supportedCards.map(card => card === 'MC' ? 'MASTERCARD' : card),
+                    billingAddressRequired: true,
+                    billingAddressParameters: {
+                        format: BillingAddressFormat.Full,
+                        phoneNumberRequired: true,
+                    },
+                },
+                tokenizationSpecification: {
+                    type: 'PAYMENT_GATEWAY',
+                    parameters: {
+                        gateway: 'stripe',
+                        'stripe:version': stripeVersion,
+                        'stripe:publishableKey': `${stripePublishableKey}/${stripeConnectedAccount}`,
+                    },
+                },
+            }],
+            transactionInfo: {
+                currencyCode,
+                totalPriceStatus: 'FINAL',
+                totalPrice: round(outstandingBalance, 2).toFixed(2),
```

Comment:
Here to round we should use the decimal precision for the currency.

Reply:
We are using outstandingBalance because thats the grandTotal - store credit